util/Pos/Util/Util.hs

```diff
@@ -287,6 +281,28 @@ multilineBounds maxSize = F.later formatList
    half = maxSize' `div` 2
    remaining = maxSize' - half
 
+-- | Catch and log an exception, then rethrow it
+logException :: LoggerName -> IO a -> IO a
+logException name = E.handleAsync (\e -> handler e >> E.throw e)
+  where
+    handler :: E.SomeException -> IO ()
+    handler = usingLoggerName name . logError . pretty
+
+-- | 'bracket' which logs given message after acquiring the resource
+-- and before calling the callback with 'Info' severity.
+bracketWithLogging ::
+       (MonadMask m, WithLogger m)
+    => Text
+    -> m a
+    -> (a -> m b)
+    -> (a -> m c)
+    -> m c
+bracketWithLogging msg acquire release = bracket acquire release . addLogging
```

Comment:
Maybe also log something before (or after) the resource is released?

Reply:
It's not essential for CSL-2167, but in general it makes sense. Will do.

Reply:
Done.